pkg/kube/labels/labels.go

```diff
@@ -33,6 +36,17 @@ var (
 	}
 )
 
+// WorkloadNameFromWorkloadEntry derives the workload name from a WorkloadEntry
+func WorkloadNameFromWorkloadEntry(name string, annos map[string]string, labels map[string]string) string {
```

Comment:
So is it annotation or label or both ?

Reply:
the name thing is a label, the existing 'workload group' annotation is an annotation though. In this PR I mistakenly made the injection template read it as an annotation which is wrong; should be a label. 

Label is selected to align with canonical-service and canonical-revision